# Data Wrangling - WeRateDogs
## by Willian Rocha

## Gather

In [178]:
import pandas as pd

In [208]:
import requests, os
#read the predicted image file
# Make directory if it doesn't already exist
folder_name = 'image_predicted'
if not os.path.exists(folder_name):
    os.makedirs(folder_name)
    
image_predicted_url  = ["https://d17h27t6h515a5.cloudfront.net/topher/2017/August/599fd2ad_image-predictions/image-predictions.tsv"]

for url in image_predicted_url:
    response = requests.get(url)
    with open(os.path.join(folder_name, url.split('/')[-1]), mode = 'wb') as file:
        file.write(response.content)
        
df_image_predicted = pd.read_csv('image_predicted/image-predictions.tsv', sep='\t')

In [205]:
#read tweet from archived tweets
df_archived_tweets = pd.read_csv('twitter-archive-enhanced.csv')

In [182]:
#ref: https://www.karambelkar.info/2015/01/how-to-use-twitters-search-rest-api-most-effectively./

import tweepy
import jsonpickle

consumer_key = '' # yeYMN5z7iF91SahJ2no08wvWA
consumer_secret = ''# 12i5mFLAVBPi3W7zG0vAGoZC3N65NsOc6n28zD3DCECBNK0wwv

access_token = '' # 3296005574-9SEv7r9yh4k9NJ0wP4iB052USAnANuLjuSJd7mb
access_token_secret = '' # gld0j1MiekNgRVGhmkRX3qRMFWfWzPkNeGXaFANzsnjwF

auth = tweepy.AppAuthHandler(consumer_key, consumer_secret)

#auth.set_access_token(access_token, access_token_secret)

api = tweepy.API(auth, wait_on_rate_limit = True, wait_on_rate_limit_notify = True)

if (not api):
    print ("Can't Authenticate")

tweet_id_list = df_archived_tweets.tweet_id.tolist()
tweet_is_list_error = []

print("Downloading tweets ")

with open('tweet_json.txt', 'w') as outfile:
    for tweet_id in tweet_id_list:
        try:
            tweet = api.get_status(tweet_id, tweet_mode='extended')
            outfile.write(jsonpickle.encode(tweet._json, unpicklable=False) + '\n')
        except tweepy.TweepError as e:
            tweet_is_list_error.append(tweet_id)

print(tweet_is_list_error.count)
print ("Downloaded tweets")

Rate limit reached. Sleeping for: 206
Rate limit reached. Sleeping for: 559


<built-in method count of list object at 0x7effc6ca3148>
Downloaded tweets


In [203]:
#reading the tweet_json.txt and creating a data frame from it
import json

df_tweet_json = []

with open('tweet_json.txt', 'r') as json_file:  
    for line in json_file:
        j_content = json.loads(line)
        df_tweet_json.append({'id': j_content['id'], 
                              'favorite_count': j_content['favorite_count'],
                              'retweet_count': j_content['retweet_count']})
        
df_tweet_json = pd.DataFrame(df_tweet_json)

## Assess

In [215]:
# predição de imagens
df_image_predicted

tweet_id                                            jpg_url  \
0     666020888022790149    https://pbs.twimg.com/media/CT4udn0WwAA0aMy.jpg   
1     666029285002620928    https://pbs.twimg.com/media/CT42GRgUYAA5iDo.jpg   
2     666033412701032449    https://pbs.twimg.com/media/CT4521TWwAEvMyu.jpg   
3     666044226329800704    https://pbs.twimg.com/media/CT5Dr8HUEAA-lEu.jpg   
4     666049248165822465    https://pbs.twimg.com/media/CT5IQmsXIAAKY4A.jpg   
5     666050758794694657    https://pbs.twimg.com/media/CT5Jof1WUAEuVxN.jpg   
6     666051853826850816    https://pbs.twimg.com/media/CT5KoJ1WoAAJash.jpg   
7     666055525042405380    https://pbs.twimg.com/media/CT5N9tpXIAAifs1.jpg   
8     666057090499244032    https://pbs.twimg.com/media/CT5PY90WoAAQGLo.jpg   
9     666058600524156928    https://pbs.twimg.com/media/CT5Qw94XAAA_2dP.jpg   
10    666063827256086533    https://pbs.twimg.com/media/CT5Vg_wXIAAXfnj.jpg   
11    666071193221509120    https://pbs.twimg.com/media/CT5cN_3WEAAlOoZ.jpg   
12    666073100786774016    https://pbs.twimg.com/media/CT5d9DZXAAALcwe.jpg   
13    666082916733198337    https://pbs.twimg.com/media/CT5m4VGWEAAtKc8.jpg   
14    666094000022159362    https://pbs.twimg.com/media/CT5w9gUW4AAsBNN.jpg   
15    666099513787052032    https://pbs.twimg.com/media/CT51-JJUEAA6hV8.jpg   
16    666102155909144576    https://pbs.twimg.com/media/CT54YGiWUAEZnoK.jpg   
17    666104133288665088    https://pbs.twimg.com/media/CT56LSZWoAAlJj2.jpg   
18    666268910803644416    https://pbs.twimg.com/media/CT8QCd1WEAADXws.jpg   
19    666273097616637952    https://pbs.twimg.com/media/CT8T1mtUwAA3aqm.jpg   
20    666287406224695296    https://pbs.twimg.com/media/CT8g3BpUEAAuFjg.jpg   
21    666293911632134144    https://pbs.twimg.com/media/CT8mx7KW4AEQu8N.jpg   
22    666337882303524864    https://pbs.twimg.com/media/CT9OwFIWEAMuRje.jpg   
23    666345417576210432    https://pbs.twimg.com/media/CT9Vn7PWoAA_ZCM.jpg   
24    666353288456101888    https://pbs.twimg.com/media/CT9cx0tUEAAhNN_.jpg   
25    666362758909284353    https://pbs.twimg.com/media/CT9lXGsUcAAyUFt.jpg   
26    666373753744588802    https://pbs.twimg.com/media/CT9vZEYWUAAlZ05.jpg   
27    666396247373291520    https://pbs.twimg.com/media/CT-D2ZHWIAA3gK1.jpg   
28    666407126856765440    https://pbs.twimg.com/media/CT-NvwmW4AAugGZ.jpg   
29    666411507551481857    https://pbs.twimg.com/media/CT-RugiWIAELEaq.jpg   
...                  ...                                                ...   
2045  886366144734445568    https://pbs.twimg.com/media/DE0BTnQUwAApKEH.jpg   
2046  886680336477933568    https://pbs.twimg.com/media/DE4fEDzWAAAyHMM.jpg   
2047  886736880519319552    https://pbs.twimg.com/media/DE5Se8FXcAAJFx4.jpg   
2048  886983233522544640    https://pbs.twimg.com/media/DE8yicJW0AAAvBJ.jpg   
2049  887101392804085760    https://pbs.twimg.com/media/DE-eAq6UwAA-jaE.jpg   
2050  887343217045368832  https://pbs.twimg.com/ext_tw_video_thumb/88734...   
2051  887473957103951883    https://pbs.twimg.com/media/DFDw2tyUQAAAFke.jpg   
2052  887517139158093824  https://pbs.twimg.com/ext_tw_video_thumb/88751...   
2053  887705289381826560    https://pbs.twimg.com/media/DFHDQBbXgAEqY7t.jpg   
2054  888078434458587136    https://pbs.twimg.com/media/DFMWn56WsAAkA7B.jpg   
2055  888202515573088257    https://pbs.twimg.com/media/DFDw2tyUQAAAFke.jpg   
2056  888554962724278272    https://pbs.twimg.com/media/DFTH_O-UQAACu20.jpg   
2057  888804989199671297    https://pbs.twimg.com/media/DFWra-3VYAA2piG.jpg   
2058  888917238123831296    https://pbs.twimg.com/media/DFYRgsOUQAARGhO.jpg   
2059  889278841981685760  https://pbs.twimg.com/ext_tw_video_thumb/88927...   
2060  889531135344209921    https://pbs.twimg.com/media/DFg_2PVW0AEHN3p.jpg   
2061  889638837579907072    https://pbs.twimg.com/media/DFihzFfXsAYGDPR.jpg   
2062  889665388333682689    https://pbs.twimg.com/media/DFi579UWsAAatzw.jpg   
2063  889880896479866881    https://pbs.twimg.com/media/DFl99B1WsA

In [222]:
df_image_predicted.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2075 entries, 0 to 2074
Data columns (total 12 columns):
tweet_id    2075 non-null int64
jpg_url     2075 non-null object
img_num     2075 non-null int64
p1          2075 non-null object
p1_conf     2075 non-null float64
p1_dog      2075 non-null bool
p2          2075 non-null object
p2_conf     2075 non-null float64
p2_dog      2075 non-null bool
p3          2075 non-null object
p3_conf     2075 non-null float64
p3_dog      2075 non-null bool
dtypes: bool(3), float64(3), int64(2), object(4)
memory usage: 152.1+ KB


In [226]:
df_image_predicted[df_image_predicted.tweet_id.duplicated()]

Empty DataFrame
Columns: [tweet_id, jpg_url, img_num, p1, p1_conf, p1_dog, p2, p2_conf, p2_dog, p3, p3_conf, p3_dog]
Index: []

In [229]:
df_image_predicted.describe()

tweet_id      img_num      p1_conf       p2_conf       p3_conf
count  2.075000e+03  2075.000000  2075.000000  2.075000e+03  2.075000e+03
mean   7.384514e+17     1.203855     0.594548  1.345886e-01  6.032417e-02
std    6.785203e+16     0.561875     0.271174  1.006657e-01  5.090593e-02
min    6.660209e+17     1.000000     0.044333  1.011300e-08  1.740170e-10
25%    6.764835e+17     1.000000     0.364412  5.388625e-02  1.622240e-02
50%    7.119988e+17     1.000000     0.588230  1.181810e-01  4.944380e-02
75%    7.932034e+17     1.000000     0.843855  1.955655e-01  9.180755e-02
max    8.924206e+17     4.000000     1.000000  4.880140e-01  2.734190e-01

In [218]:
# df_archived_tweets
df_archived_tweets

tweet_id  in_reply_to_status_id  in_reply_to_user_id  \
0     892420643555336193                    NaN                  NaN   
1     892177421306343426                    NaN                  NaN   
2     891815181378084864                    NaN                  NaN   
3     891689557279858688                    NaN                  NaN   
4     891327558926688256                    NaN                  NaN   
5     891087950875897856                    NaN                  NaN   
6     890971913173991426                    NaN                  NaN   
7     890729181411237888                    NaN                  NaN   
8     890609185150312448                    NaN                  NaN   
9     890240255349198849                    NaN                  NaN   
10    890006608113172480                    NaN                  NaN   
11    889880896479866881                    NaN                  NaN   
12    889665388333682689                    NaN                  NaN   
13    889638837579907072                    NaN                  NaN   
14    889531135344209921                    NaN                  NaN   
15    889278841981685760                    NaN                  NaN   
16    888917238123831296                    NaN                  NaN   
17    888804989199671297                    NaN                  NaN   
18    888554962724278272                    NaN                  NaN   
19    888202515573088257                    NaN                  NaN   
20    888078434458587136                    NaN                  NaN   
21    887705289381826560                    NaN                  NaN   
22    887517139158093824                    NaN                  NaN   
23    887473957103951883                    NaN                  NaN   
24    887343217045368832                    NaN                  NaN   
25    887101392804085760                    NaN                  NaN   
26    886983233522544640                    NaN                  NaN   
27    886736880519319552                    NaN                  NaN   
28    886680336477933568                    NaN                  NaN   
29    886366144734445568                    NaN                  NaN   
...                  ...                    ...                  ...   
2326  666411507551481857                    NaN                  NaN   
2327  666407126856765440                    NaN                  NaN   
2328  666396247373291520                    NaN                  NaN   
2329  666373753744588802                    NaN                  NaN   
2330  666362758909284353                    NaN                  NaN   
2331  666353288456101888                    NaN                  NaN   
2332  666345417576210432                    NaN                  NaN   
2333  666337882303524864                    NaN                  NaN   
2334  666293911632134144                    NaN                  NaN   
2335  666287406224695296                    NaN                  NaN   
2336  666273097616637952                    NaN                  NaN   
2337  666268910803644416                    NaN                  NaN   
2338  666104133288665088                    NaN                  NaN   
2339  666102155909144576                    NaN                  NaN   
2340  666099513787052032                    NaN                  NaN   
2341  666094000022159362                    NaN                  NaN   
2342  666082916733198337                    NaN                  NaN   
2343  666073100786774016                    NaN                  NaN   
2344  666071193221509120                    NaN                  NaN   
2345  666063827256086533                    NaN                  NaN   
2346  666058600524156928                    NaN                  NaN   
2347  666057090499244032                    NaN                  NaN   
2348  666055525042405380                    NaN                  NaN   
2349  666051853826850816                    NaN         

In [224]:
df_archived_tweets.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2356 entries, 0 to 2355
Data columns (total 17 columns):
tweet_id                      2356 non-null int64
in_reply_to_status_id         78 non-null float64
in_reply_to_user_id           78 non-null float64
timestamp                     2356 non-null object
source                        2356 non-null object
text                          2356 non-null object
retweeted_status_id           181 non-null float64
retweeted_status_user_id      181 non-null float64
retweeted_status_timestamp    181 non-null object
expanded_urls                 2297 non-null object
rating_numerator              2356 non-null int64
rating_denominator            2356 non-null int64
name                          2356 non-null object
doggo                         2356 non-null object
floofer                       2356 non-null object
pupper                        2356 non-null object
puppo                         2356 non-null object
dtypes: float64(4), int64(3), ob

In [230]:
df_archived_tweets[df_archived_tweets.tweet_id.duplicated()]

Empty DataFrame
Columns: [tweet_id, in_reply_to_status_id, in_reply_to_user_id, timestamp, source, text, retweeted_status_id, retweeted_status_user_id, retweeted_status_timestamp, expanded_urls, rating_numerator, rating_denominator, name, doggo, floofer, pupper, puppo]
Index: []

In [232]:
df_archived_tweets.sample(5)

tweet_id  in_reply_to_status_id  in_reply_to_user_id  \
1353  703769065844768768                    NaN                  NaN   
1328  705970349788291072                    NaN                  NaN   
1197  717009362452090881                    NaN                  NaN   
1931  674036086168010753                    NaN                  NaN   
269   841320156043304961                    NaN                  NaN   

                      timestamp  \
1353  2016-02-28 02:29:55 +0000   
1328  2016-03-05 04:17:02 +0000   
1197  2016-04-04 15:22:08 +0000   
1931  2015-12-08 01:21:40 +0000   
269   2017-03-13 16:08:50 +0000   

                                                 source  \
1353  <a href="http://twitter.com/download/iphone" r...   
1328  <a href="http://twitter.com/download/iphone" r...   
1197  <a href="http://twitter.com/download/iphone" r...   
1931  <a href="http://twitter.com/download/iphone" r...   
269   <a href="http://twitter.com/download/iphone" r...   

                                                   text  retweeted_status_id  \
1353  When you're trying to watch your favorite tv s...                  NaN   
1328  This is Lucy. She's a Venetian Kerploof. Suppo...                  NaN   
1197  This is Smokey. He's having some sort of exist...                  NaN   
1931  Meet Daisy. She has no eyes &amp; her face has...                  NaN   
269   We don't rate penguins, but if we did, this on...                  NaN   

      retweeted_status_user_id retweeted_status_timestamp  \
1353                       NaN                        NaN   
1328                       NaN                        NaN   
1197                       NaN                        NaN   
1931                       NaN                        NaN   
269                        NaN                        NaN   

                                          expanded_urls  rating_numerator  \
1353  https://twitter.com/dog_rates/status/703769065...                10   
1328  https://twitter.com/dog_rates/status/705970349...                12   
1197  https://twitter.com/dog_rates/status/717009362...                10   
1931  https://twitter.com/dog_rates/status/674036086...                 9   
269   https://twitter.com/abc/status/841311395547250688                12   

      rating_denominator    name doggo floofer  pupper puppo  
1353                  10    None  None    None    None  None  
1328                  10    Lucy  None    None    None  None  
1197                  10  Smokey  None    None  pupper  None  
1931                  10   Daisy  None    None    None  None  
269                   10    None  None    None    None  None

In [235]:
df_archived_tweets.rating_numerator.value_counts()

12      558
11      464
10      461
13      351
9       158
8       102
7        55
14       54
5        37
6        32
3        19
4        17
1         9
2         9
420       2
0         2
15        2
75        2
80        1
20        1
24        1
26        1
44        1
50        1
60        1
165       1
84        1
88        1
144       1
182       1
143       1
666       1
960       1
1776      1
17        1
27        1
45        1
99        1
121       1
204       1
Name: rating_numerator, dtype: int64

In [246]:
df_archived_tweets.loc[df_archived_tweets.rating_numerator == 420]

tweet_id  in_reply_to_status_id  in_reply_to_user_id  \
188   855862651834028034           8.558616e+17          194351775.0   
2074  670842764863651840                    NaN                  NaN   

                      timestamp  \
188   2017-04-22 19:15:32 +0000   
2074  2015-11-29 05:52:33 +0000   

                                                 source  \
188   <a href="http://twitter.com/download/iphone" r...   
2074  <a href="http://twitter.com/download/iphone" r...   

                                                   text  retweeted_status_id  \
188   @dhmontgomery We also gave snoop dogg a 420/10...                  NaN   
2074  After so many requests... here you go.\n\nGood...                  NaN   

      retweeted_status_user_id retweeted_status_timestamp  \
188                        NaN                        NaN   
2074                       NaN                        NaN   

                                          expanded_urls  rating_numerator  \
188                                                 NaN               420   
2074  https://twitter.com/dog_rates/status/670842764...               420   

      rating_denominator  name doggo floofer pupper puppo  
188                   10  None  None    None   None  None  
2074                  10  None  None    None   None  None

In [285]:
df_archived_tweets[df_archived_tweets.retweeted_status_id.isnull() & df_archived_tweets.in_reply_to_status_id.isnull()].sample(5)

tweet_id  in_reply_to_status_id  in_reply_to_user_id  \
808   771770456517009408                    NaN                  NaN   
1190  718234618122661888                    NaN                  NaN   
243   846139713627017216                    NaN                  NaN   
2072  671109016219725825                    NaN                  NaN   
533   807621403335917568                    NaN                  NaN   

                      timestamp  \
808   2016-09-02 18:03:10 +0000   
1190  2016-04-08 00:30:51 +0000   
243   2017-03-26 23:20:02 +0000   
2072  2015-11-29 23:30:32 +0000   
533   2016-12-10 16:22:02 +0000   

                                                 source  \
808   <a href="http://twitter.com/download/iphone" r...   
1190  <a href="http://twitter.com/download/iphone" r...   
243   <a href="http://twitter.com/download/iphone" r...   
2072  <a href="http://twitter.com/download/iphone" r...   
533   <a href="http://twitter.com/download/iphone" r...   

                                                   text  retweeted_status_id  \
808   This is Davey. He'll have your daughter home b...                  NaN   
1190  This is Suki. She was born with a blurry tail ...                  NaN   
243   SHE DID AN ICY ZOOM AND KNEW WHEN TO PUT ON TH...                  NaN   
2072  This is Toby. He asked for chocolate cake for ...                  NaN   
533   This is Ollie Vue. He was a 3 legged pupper on...                  NaN   

      retweeted_status_user_id retweeted_status_timestamp  \
808                        NaN                        NaN   
1190                       NaN                        NaN   
243                        NaN                        NaN   
2072                       NaN                        NaN   
533                        NaN                        NaN   

                                          expanded_urls  rating_numerator  \
808   https://twitter.com/dog_rates/status/771770456...                11   
1190  https://twitter.com/dog_rates/status/718234618...                11   
243   https://twitter.com/csncapitals/status/8460884...                13   
2072  https://twitter.com/dog_rates/status/671109016...                 8   
533   https://twitter.com/dog_rates/status/807621403...                14   

      rating_denominator   name doggo floofer  pupper puppo  
808                   10  Davey  None    None    None  None  
1190                  10   Suki  None    None    None  None  
243                   10   None  None    None    None  None  
2072                  10   Toby  None    None    None  None  
533                   10  Ollie  None    None  pupper  None

In [284]:
df_archived_tweets.loc[df_archived_tweets.rating_numerator == 14]

tweet_id  in_reply_to_status_id  in_reply_to_user_id  \
9     890240255349198849                    NaN                  NaN   
22    887517139158093824                    NaN                  NaN   
35    885518971528720385                    NaN                  NaN   
41    884441805382717440                    NaN                  NaN   
56    881536004380872706                    NaN                  NaN   
64    879674319642796034           8.795538e+17         3.105441e+09   
68    879130579576475649                    NaN                  NaN   
76    878057613040115712                    NaN                  NaN   
78    877611172832227328                    NaN                  NaN   
83    876537666061221889                    NaN                  NaN   
95    873697596434513921                    NaN                  NaN   
101   872668790621863937                    NaN                  NaN   
110   871102520638267392                    NaN                  NaN   
117   870063196459192321                    NaN                  NaN   
123   868880397819494401                    NaN                  NaN   
141   864873206498414592                    NaN                  NaN   
149   863079547188785154           6.671522e+17         4.196984e+09   
164   860184849394610176                    NaN                  NaN   
184   856526610513747968           8.558181e+17         4.196984e+09   
185   856330835276025856                    NaN                  NaN   
186   856288084350160898           8.562860e+17         2.792810e+08   
187   856282028240666624                    NaN                  NaN   
195   855138241867124737                    NaN                  NaN   
199   854120357044912130                    NaN                  NaN   
209   852226086759018497                    NaN                  NaN   
214   851464819735769094                    NaN                  NaN   
268   841439858740625411                    NaN                  NaN   
298   837012587749474308                    NaN                  NaN   
339   832273440279240704                    NaN                  NaN   
347   831911600680497154                    NaN                  NaN   
369   828650029636317184                    NaN                  NaN   
372   828381636999917570                    NaN                  NaN   
395   825535076884762624                    NaN                  NaN   
404   824025158776213504                    NaN                  NaN   
418   822462944365645825                    NaN                  NaN   
426   821407182352777218                    NaN                  NaN   
434   820494788566847489                    NaN                  NaN   
436   820314633777061888                    NaN                  NaN   
446   819015337530290176                    NaN                  NaN   
447   819015331746349057                    NaN                  NaN   
448   819006400881917954                    NaN                  NaN   
449   819004803107983360                    NaN                  NaN   
490   813812741911748608                    NaN                  NaN   
533   807621403335917568                    NaN                  NaN   
589   799308762079035393                    NaN                  NaN   
631   794205286408003585                    NaN                  NaN   
657   791774931465953280                    NaN                  NaN   
758   778408200802557953                    NaN                  NaN   
788   774314403806253056                    NaN                  NaN   
866   762035686371364864                    NaN                  NaN   
924   755206590534418437                    NaN                  NaN   
1053  742465774154047488                    NaN                  NaN   
1605  685681090388975616           6.855479e+17         4.196984e+09   
1609  685547936038666240                    NaN                  NaN   

                      timestamp  \
9     2017-07-26 15:

In [241]:
df_archived_tweets.rating_denominator.value_counts()

10     2333
11        3
50        3
80        2
20        2
2         1
16        1
40        1
70        1
15        1
90        1
110       1
120       1
130       1
150       1
170       1
7         1
0         1
Name: rating_denominator, dtype: int64

In [243]:
df_archived_tweets.loc[df_archived_tweets.rating_denominator == 70]

tweet_id  in_reply_to_status_id  in_reply_to_user_id  \
433  820690176645140481                    NaN                  NaN   

                     timestamp  \
433  2017-01-15 17:52:40 +0000   

                                                source  \
433  <a href="http://twitter.com/download/iphone" r...   

                                                  text  retweeted_status_id  \
433  The floofs have been released I repeat the flo...                  NaN   

     retweeted_status_user_id retweeted_status_timestamp  \
433                       NaN                        NaN   

                                         expanded_urls  rating_numerator  \
433  https://twitter.com/dog_rates/status/820690176...                84   

     rating_denominator  name doggo floofer pupper puppo  
433                  70  None  None    None   None  None

In [220]:
df_tweet_json

favorite_count                  id  retweet_count
0              38377  892420643555336193           8431
1              32885  892177421306343426           6211
2              24780  891815181378084864           4111
3              41719  891689557279858688           8553
4              39887  891327558926688256           9284
5              20020  891087950875897856           3081
6              11719  890971913173991426           2045
7              64781  890729181411237888          18691
8              27531  890609185150312448           4223
9              31597  890240255349198849           7319
10             30344  890006608113172480           7260
11             27498  889880896479866881           4927
12             47625  889665388333682689           9961
13             26891  889638837579907072           4494
14             14939  889531135344209921           2214
15             25036  889278841981685760           5364
16             28790  888917238123831296           4455
17             25326  888804989199671297           4278
18             19672  888554962724278272           3539
19             21531  888078434458587136           3464
20             29849  887705289381826560           5332
21             45779  887517139158093824          11549
22             68387  887473957103951883          18015
23             33329  887343217045368832          10298
24             30242  887101392804085760           5907
25             34796  886983233522544640           7703
26             11930  886736880519319552           3258
27             22189  886680336477933568           4420
28             20975  886366144734445568           3171
29               116  886267009285017600              4
...              ...                 ...            ...
2311             442  666411507551481857            324
2312             108  666407126856765440             40
2313             167  666396247373291520             84
2314             187  666373753744588802             92
2315             774  666362758909284353            566
2316             219  666353288456101888             72
2317             293  666345417576210432            135
2318             197  666337882303524864             90
2319             498  666293911632134144            352
2320             146  666287406224695296             64
2321             174  666273097616637952             76
2322             102  666268910803644416             35
2323           14225  666104133288665088           6530
2324              80  666102155909144576             12
2325             153  666099513787052032             67
2326             163  666094000022159362             73
2327             117  666082916733198337             44
2328             320  666073100786774016            163
2329             146  666071193221509120             60
2330             471  666063827256086533            216
2331             111  666058600524156928             57
2332             295  666057090499244032            141
2333             433  666055525042405380            247
2334            1210  666051853826850816            843
2335             132  666050758794694657             59
2336             109  666049248165822465             40
2337             298  666044226329800704            140
2338             125  666033412701032449             44
2339             128  666029285002620928             47
2340            2554  666020888022790149            508

[2341 rows x 3 columns]

In [221]:
df_tweet_json.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2341 entries, 0 to 2340
Data columns (total 3 columns):
favorite_count    2341 non-null int64
id                2341 non-null int64
retweet_count     2341 non-null int64
dtypes: int64(3)
memory usage: 54.9 KB


#### Quality
##### `image predicted`
- Quantidade de Tweets (2075) é diferente da base de Tweet arquivados (2356).

#### `archived tweets`
- Falta do nome do cachorro em alguns tweets.
- Falta da classicação do estágio do cachorro (doggo / floofer / pupper / puppo) em alguns tweets.
- rating_numerator muito alto em alguns casos.
- rating_denominator maior que 10 (alguns tweets com mais de 1 cachorro, ex.: 820690176645140481).
- Falta da coluna Quantidade de Favorito.
- Falta da Coluna Quantidade de retweet.
- Alguns tweets da coluna nao estao corretos, por exemplo, such, this.

#### Tidiness

## Clean

#### Define

#### Code

#### Test